lib/common_test/doc/src/ct_suite.xml

```diff
@@ -0,0 +1,619 @@
+    </datatype>
+    <datatype>
+      <name name="ct_config" n_vars="0"/>
+      <desc><p>The configuration data that can be modified.</p></desc>
+    </datatype>
+    <datatype>
+      <name name="ct_status" n_vars="0"/>
+      <desc><p>The status value for a nested subgroup.</p></desc>
+    </datatype>
+    <datatype>
+      <name>ct_group_def()</name>
+      <desc><p>The test group definition, as returned by <seemfa marker="#Module:groups/0"><c>Module:groups/0</c></seemfa>.</p></desc>
+    </datatype>
+    <datatype>
+      <name>ct_test_def()</name>
+      <desc><p>The test suite definition, as returned by <seemfa marker="#Module:all/0"><c>Module:all/0</c></seemfa>.</p></desc>
+    </datatype>
+    <datatype>
+      <name>ct_info()</name>
+      <desc><p>The test suite information, as returned by <seemfa marker="#Module:suite/0"><c>Module:suite/0</c></seemfa>, <seemfa marker="#Module:group/1"><c>Module:group/1</c></seemfa> and <seemfa marker="#Module:Testcase/0"><c>Module:Testcase/0</c></seemfa>.</p></desc>
+    </datatype>
+  </datatypes>
+
+  <funcs>
+    <fsdescription>
+      <title>Callback Functions</title>
+      <p>
+        The following functions are to be exported from a
+        <c>ct_suite</c> callback module in order to define
+        the callback interface for a test suite.
+      </p>
+    </fsdescription>
+
+    <func>
+      <name since="">Module:all() -> [ct_test_def()] | {skip, Reason}</name>
+      <fsummary>Returns the list of all test case groups and test cases
+	in the module.</fsummary>
+      <type>
+	<v><seetype marker="#ct_test_def">ct_test_def()</seetype> = TestCase | {group, GroupName} | {group, GroupName, Properties} | {group, GroupName, Properties, SubGroups}</v>
+	<v>TestCase = <seetype marker="#ct_testname">ct_testname()</seetype></v>
+	<v>GroupName = <seetype marker="#ct_groupname">ct_groupname()</seetype></v>
+	<v>Properties = [parallel | sequence | Shuffle | {RepeatType, N}] | default</v>
+	<v>SubGroups = [{GroupName, Properties} | {GroupName, Properties, SubGroups}]</v>
+	<v>Shuffle = shuffle | {shuffle, Seed}</v>
+	<v>Seed = {integer(), integer(), integer()}</v>
+	<v>RepeatType = repeat | repeat_until_all_ok | repeat_until_all_fail | repeat_until_any_ok | repeat_until_any_fail</v>
+	<v>N = integer() | forever</v>
+	<v>Reason = term()</v>
+      </type>
+
+      <desc>
+	<p>MANDATORY</p>
+
+	<p>Returns the list of all test cases and test case groups in the
+          test suite module to be executed. This list also specifies the
+          order the cases and groups are executed by <c>Common Test</c>.
+          A test case is represented by an atom,
+	  the name of the test case function, or a <c>testcase</c> tuple
+	  indicating that the test case shall be repeated. A test case group is
+	  represented by a <c>group</c> tuple, where <c>GroupName</c>,
+	  an atom, is the name of the group (defined in
+	  <seemfa marker="#Module:groups/0"><c>Module:groups/0</c></seemfa>).
```

Comment:
It would be better to use  ```<seetype>``` tags instead of  ```<seemfa>``` when referring to types throughout this document. An example from #2793 :
```xml
<seetype marker="uri_string#uri_string"><c>uri_string()</c></seetype>
```

Reply:
I can do that, but most of those tags were copied from common_test_app.xml.

Reply:
I can't see a misuse of `seemfa`: will look for misuse of `seetype`. The place where you add the comment seems to be as expected, since `groups/0` is a function, not a type.

Reply:
I can't see a misuse of `seetype` either.
Where did you find an issue, specifically?

Reply:
You are right, please disregard this comment.